src/cmap/wire_protocol/compression.ts

```diff
@@ -39,12 +39,20 @@ export function compress(
 ): void {
   const zlibOptions = {} as zlib.ZlibOptions;
   switch (self.options.agreedCompressor) {
-    case 'snappy':
+    case 'snappy': {
       if ('kModuleError' in Snappy) {
         return callback(Snappy['kModuleError']);
       }
-      Snappy.compress(dataToBeCompressed, callback);
+
+      const snappyResult = Snappy.compress(dataToBeCompressed, callback);
```

Comment:
Here's my solution to NODE-3528, should we check the function's length here to be more sure?